src/zns.ts

```diff
@@ -154,6 +156,40 @@ export default class Zns extends NamingService {
     return this.structureResolverRecords(await this.records(domain));
   }
 
+  /**
+   * Resolves the ipfs hash stored in domain record
+   * @param domain - domain name
+   * @throws ResolutionError with 
+   *  - NoIPFSConfigurationFound code when records has no fields for ipfs
+   *  - UnregisteredDomain code when no records was found
+   * @returns a promise that resolves in a value stored under ipfs.html.value or undefined. 
+   */
+  async ipfsHash(domain: string): Promise<string> {
+    const record = await this.structureResolverRecords(await this.records(domain));
+    if (!record)
+      throw new ResolutionError('UnregisteredDomain', {domain});
+    if (!record.ipfs || !record.ipfs.html)
+      throw new ResolutionError('NoIPFSConfigurationFound', {domain});
+    return record.ipfs.html.value;
+  }
+
+  /**
+   * Resolves ipfs whois object stored in domain record
+   * @param domain - domain name
+   * @throws ResolutionError with 
+   *  - UncofiguredWhoIs code when no such filed found in records
+   *  - UnregisteredDomain code when no records was found
+   */
+  async ipfsWhois(domain: string): Promise<WhoIsStructure> {
```

Comment:
I don't see `whois` as related to `ipfs` by any reason. Also, CNS would not allow to query several records at the same time. I think we can only have records queried one by one at the moment. So lets have `email` method for that. We do not need `for_sale` at the moment and I am not sure it is appropriate without pricing information.